ClosedXML/Excel/CalcEngine/XLCalcEngine.cs

```diff
@@ -23,6 +23,50 @@ public XLCalcEngine(IXLWorksheet ws) : this(ws.Workbook)
             _ws = ws;
         }
 
+        private IList<IXLRange> _cellRanges;
+
+        /// <summary>
+        /// Get a collection of cell ranges included into the expression. Order is not preserved.
+        /// </summary>
+        /// <param name="expression">Formula to parse.</param>
+        /// <returns>Collection of ranges included into the expression.</returns>
+        public IEnumerable<IXLRange> GetAffectingRanges(string expression)
+        {
+            _cellRanges = new List<IXLRange>();
+            Parse(expression);
+            var ranges = _cellRanges;
+            _cellRanges = null;
+            var visitedRanges = new HashSet<IXLRangeAddress>(new XLRangeAddressComparer(true));
+            foreach (var range in ranges)
+            {
+                if (!visitedRanges.Contains(range.RangeAddress))
+                {
+                    visitedRanges.Add(range.RangeAddress);
+                    yield return range;
+                }
+            }
+        }
+
+        public IEnumerable<IXLCell> GetAffectingCells(string expression)
```

Comment:
Please rename to `GetPrecedentCells`.

By the way, we might want to expose this method publicly on the API. Could be useful. Leave for later.